CMakeLists.txt

```diff
@@ -90,15 +91,25 @@ if(TARGET Boost::program_options)
     target_link_libraries(sphinxsys_core INTERFACE Boost::program_options)
 endif()
 
+if(SPHINXSYS_USE_SYCL)
+    if(NOT SPHINXSYS_SYCL_TARGETS)
+        set(SPHINXSYS_SYCL_TARGETS nvptx64-nvidia-cuda)
+    endif()
+    message("-- SPHinXsys use SYCL with targets ${SPHINXSYS_SYCL_TARGETS}")
+    target_compile_options(sphinxsys_core INTERFACE -fsycl -fsycl-targets=${SPHINXSYS_SYCL_TARGETS})
+    target_link_options(sphinxsys_core INTERFACE -fsycl -fsycl-targets=${SPHINXSYS_SYCL_TARGETS})
```

Comment:
Also are all flags the same for all implementation of SYCL, and all with the same (GCC/Clang) format ?

Reply:
Yes, different implementations have different interfaces to CMake and their compilers. For example, [Open SYCL](https://github.com/OpenSYCL/OpenSYCL) (aka hipSYCL) has a find directive for CMake.

For what regards DPC++ (and the original question of @ChiZhangatTUM) [in their documentation](https://intel.github.io/llvm-docs/GetStartedGuide.html#build-dpc-application-with-cmake) they are suggesting to use both `target_compile_options` and `target_link_options` or using `CMAKE_CXX_FLAGS` directly. 

What I did in a previous project was to have a CMake option (something on the line of `-DSYCL_IMPL=dpcpp`) and then having an if/else switch in CMakeLists to import the selected implementation as needed. This obviously requires us to independently support each implementation.

Reply:
Another thing is that, I am wondering whether we can use icpx from the intel base toolkit directly other than use dpcpp which is said to be deprecated.

Reply:
I can confirm that icpx works as expected.

> `-DSYCL_IMPL=dpcpp`

Here I meant that we are selecting DPC++ instead of other SYCL implementations, but the binary itself has been clang++ so far.

Reply:
> 

I also tried with OneAPI toolkit and Nvidia SYCL backend, the present branch works fine with clang++ compiler. 
I will try icpx compiler for clarity.